Production/src/Selectors.cc

```diff
@@ -174,7 +192,104 @@ TauJetSelector::Result TauJetTag::Select(const edm::Event& event, const std::deq
+            if(!(electron.pt() > 34 && std::abs(electron.eta()) < 2.3 && electron.electronID("mvaEleID-Fall17-noIso-V2-wp90") > 0.5f && PFRelIsolation(electron,rho) < 0.15
+                     && std::abs(electron.bestTrack()->dz(primaryVertex.position())) < 0.2
+                     && std::abs(electron.bestTrack()->dxy(primaryVertex.position())) < 0.045))
+            	 continue;
+            if(!ref_electron || PFRelIsolation(*ref_electron,rho) > PFRelIsolation(electron,rho) || (PFRelIsolation(*ref_electron,rho) == PFRelIsolation(electron,rho) && ref_electron->pt() < electron.pt())){
+                ref_electron = &electron;
+            }
+        }
+    }
+    //
+    auto tagObject = std::make_shared<TagObject>();
+    std::vector<const TauJet*> selectedTauJets;
+    std::vector<Type> selectedTypes;
+    if(ref_muon || ref_electron || met.pt() > 180) {
+    	if(ref_muon) { // Fill tag object with muon if muon found
+        	tagObject->kind = tau_analysis::selectors::TagObject::Kind::Muon; //new tag object member 
+        	tagObject->p4 = ref_muon->polarP4();
+       		tagObject->charge = ref_muon->charge();
+        	tagObject->id = unsigned(ref_muon->isLooseMuon()) * 1 + unsigned(ref_muon->isMediumMuon()) * 2
+                          	+ unsigned(ref_muon->isTightMuon(primaryVertex)) * 4;
+        	tagObject->isolation = PFRelIsolation(*ref_muon);
+        	tagObject->has_extramuon = hasExtraMuon(muons, ref_muon, primaryVertex);
+        	tagObject->has_extraelectron = hasExtraElectron(electrons, ref_electron, rho);
+        	tagObject->has_dimuon = hasExtraDimuon(muons, primaryVertex);
+		tagObject->has_dielectron = hasExtraDielectron(electrons, rho);
+    	}
+    	//	
+    	else if(ref_electron) { // Fill tag with ele if ele found rather than muon
+		tagObject->kind = tau_analysis::selectors::TagObject::Kind::Electron; //new tag object member
+        	tagObject->p4 = ref_electron->polarP4();
+        	tagObject->charge = ref_electron->charge();
+        	tagObject->id = ref_electron->electronID("mvaEleID-Fall17-noIso-V2-wp90"); //Not sure what to put here 
+        	tagObject->isolation = PFRelIsolation(*ref_electron,rho);
+        	tagObject->has_extramuon = hasExtraMuon(muons, ref_muon, primaryVertex);
+        	tagObject->has_extraelectron = hasExtraElectron(electrons, ref_electron, rho);
+        	tagObject->has_dimuon = hasExtraDimuon(muons, primaryVertex);	
+		tagObject->has_dielectron = hasExtraDielectron(electrons, rho);
+    	}
+    	else { // high MET case
+        	tagObject = nullptr;   
+    	}
+        const TauJet* selected = nullptr;
+        for(const TauJet& tauJet : tauJets) { //Look for highest pT jet
+            if(!tauJet.jet) continue;
+            const pat::Jet& jet = *tauJet.jet;
+            if(!(std::abs(jet.eta()) < 2.8) || ((ref_muon || ref_electron) && !(reco::deltaR(jet.p4(), tagObject->p4) > 0.8)))
```

Comment:
I was thinking about deltaR requirement. Maybe 0.8 is too much: current deepTau training still show good performance for small deltaR, so I think it is safe to go to 0.4. Would you agree?